.github/actions/initialize/action.yml

```diff
@@ -0,0 +1,19 @@
+name: 'Initialize'
+description: 'Checkout repo and install dependencies'
+runs:
+  using: "composite"
+  steps:
+    - name: Check out repository code
+      uses: actions/checkout@v4
+    - name: Upgrade pip
+      run: python -m pip install --upgrade pip
+      shell: bash
+    - name: Install runtime dependencies
+      run: python -m pip install --only-binary ':all:' -r requirements.txt --upgrade
+      shell: bash
+    - name: Install test dependencies
+      run: python -m pip install --only-binary ':all:' -r test_requirements.txt --upgrade
+      shell: bash
+    - name: Install citrine
+      run: python -m pip install --only-binary ':all:' -e . --upgrade
+      shell: bash
```

Comment:
What is the `--upgrade` supposed to accomplish?  All the entries in requirements.txt are pinned, so that flag has no impact.

Reply:
The safety of not knowing if some of the more common packages we use (e.g. `requests`, `pytest`) are already installed at different versions.

Reply:
(For posterity, since we chatting in Slack)

I tried it again locally, and realized my testing was screwy. You are correct, it's not needed except for when we don't specify the package version, such as the pip upgrade line.